bedrock/mozorg/management/commands/update_product_details_files.py

```diff
@@ -0,0 +1,93 @@
+
+FIREFOX_VERSION_KEYS = (
+    'FIREFOX_NIGHTLY',
+    'FIREFOX_AURORA',
+    'FIREFOX_ESR',
+    'FIREFOX_ESR_NEXT',
+    'LATEST_FIREFOX_DEVEL_VERSION',
+    'LATEST_FIREFOX_RELEASED_DEVEL_VERSION',
+    'LATEST_FIREFOX_VERSION',
+)
+
+
+class Command(BaseCommand):
+
+    def __init__(self, stdout=None, stderr=None, no_color=False):
+        self.file_storage = PDFileStorage()
+        self.db_storage = PDDatabaseStorage()
+        super(Command, self).__init__(stdout, stderr, no_color)
+
+    def add_arguments(self, parser):
+        parser.add_argument('-f', '--force', action='store_true', dest='force', default=False,
+                            help=('Download product details even if they have '
+                                  'not been updated since the last fetch.'))
+
+        parser.add_argument('-q', '--quiet', action='store_true', dest='quiet', default=False,
+                            help='If no error occurs, swallow all output.'),
+        parser.add_argument('--database', default='default',
+                            help=('Specifies the database to use, if using a db. '
+                                  'Defaults to "default".')),
+
+    def handle(self, *args, **options):
+        if not settings.PROD_DETAILS_STORAGE.endswith('PDDatabaseStorage'):
+            raise CommandError('Must be setup for database product-details storage to use this')
+
+        self.update_file_data(options)
+        try:
+            self.validate_data()
+        except Exception:
+            raise CommandError('Product Details data is invalid')
+
+        if not options['quiet']:
+            print('Product Details data is valid')
+
+        self.load_changes(options)
+
+        if not options['quiet']:
+            print('Product Details data update is complete')
+
+    def load_changes(self, options):
+        with transaction.atomic(using=options['database']):
+            for filename in self.file_storage.all_json_files():
+                fs_file_mtime = self.file_storage.last_modified_datetime(filename)
+                db_file_mtime = self.db_storage.last_modified_datetime(filename)
+                if options['force'] or not db_file_mtime or fs_file_mtime > db_file_mtime:
+                    self.db_storage.update(filename,
+                                           self.file_storage.content(filename),
+                                           self.file_storage.last_modified(filename))
+                    if not options['quiet']:
+                        print('Updated ' + filename)
+
+            self.db_storage.update('/', '', self.file_storage.last_modified('/'))
+            self.db_storage.update('regions/', '', self.file_storage.last_modified('regions/'))
+
+    def update_file_data(self, options):
+        # json dir set in settings
+        command = PDCommand()
+        command._storage = self.file_storage
+        command.is_db_storage = False
+        command.handle(**options)
```

Comment:
I know this whole thing is supposed to be a quick hack that we plan on replacing soon, and you would normally take a cleaner, less brittle approach, but I feel obligated to point out that we really shouldn't do this under normal circumstances.